proto/addressmetadata.proto

```diff
@@ -1,10 +1,34 @@
 syntax = "proto3";
 package models;
 
-// Basic key/value used to store header data.
-message Header {
-    string name = 1;
-    string value = 2;
+// Specifies the revocation method
+enum RevocationMethod {
+    /* If any of the pubkeys given in commitment_data[..n-1] signs
+    data commitment_data[n] then revocation is granted. */
+    BackupPubkey = 0;
+}
+
+// Message sent to server to revoke a key
+message Revocation {
+    // The revocation method
+    RevocationMethod method = 1;
+    // Data associated with the reveal phase
+    repeated bytes reveal_data = 2;
+    // New key to redirect to
+    bytes pub_key_hash = 3;
```

Comment:
I'm not sure this should be in here.  It's specific enough that it should be part of the revocation data itself.